app/lib/git_repository.rb

```diff
@@ -38,17 +38,29 @@ def initialize(connect_string)
         @repos.reset('origin/master', :hard) # align to whatever is in origin/master
       rescue StandardError # TODO this shouldn't be necessary. It catches the case when a local repo is corrupted,
                            #      we need to prevent the corruption instead/as well
+        GitRepository.reclone_repo(connect_string)
+      end
+    else
+      GitRepository.reclone_repo(connect_string)
```

Comment:
This is the only relevant comment, everything else is about style:

This makes it asymmetric with svn and memory, where we raise an exception instead.
I know what you're trying to do, but it would mean that even with an invalid connect_string (i.e. a repository that is not supposed to exist) you would try to reclone the repo over and over. Is this supposed to go away after we land a proper concurrency fix?

Reply:
To your last question: yes (see the last paragraph of the original comment on the PR)

However, I'll update it so that it doesn't try to reclone if the bare repository doesn't exist for now as well.

Reply:
Perfect, that'll do!